### ISSUE TITLE
Request: $Values to type something as the union of values of an object

### ISSUE THREAD

**Issue by kylehg** [facebook/flow#3487]
**TL;DR** It would be great to add a type function that expresses "the union of all the values of a static object," i.e., like `$Keys` but for `$Values`.

I'm working in a codebase that has been written with an eye toward Closure Compiler, and makes extensive use of the following enum pattern:

```js
/** @enum {string} */
const ButterBarType = {
  SUCCESS: 'success',
  ERROR: 'error',
}

activateButterBar("It worked!", ButterBarType.SUCCESS)
activateButterBar("It failed :(", ButterBarType.ERROR)
```

Besides being Closure Compiler–enforced, this pattern is useful for expressing in code a limited set of possible values, and is also great for interacting with data storage, since we can adapt our naming as code might change, but leave the values stored the same. Sadly, though, Flow offers no expressive way to write the annotation for `activateButterBar` without either defaulting to the overly generic `string` or manually adding `type ButterBarTypeT = 'success' | 'error'` and keeping it in sync.

Obviously a `$Values` would have trouble with dynamic values, but figure it's no worse than `$Keys`, and the compiler could reject the typing for values that can't be statically inferred.

**Comment by nmn** [comment]
Dupe of many other issues.

**Comment by AlicanC** [comment]
FYI, https://github.com/facebook/flow/issues/961 is open.
